source/Octopus.Server.Client/Exceptions/OctopusExceptionFactory.cs

```diff
@@ -51,74 +51,96 @@ public static async Task<OctopusException> CreateException(HttpResponseMessage r
-                    {
-                        fullDetails += Environment.NewLine + "Server exception: " + Environment.NewLine + errors.FullException + Environment.NewLine + "-----------------------" + Environment.NewLine;
-                    }
-                }
-            }
-            // ReSharper disable once EmptyGeneralCatchClause
-            catch
+                HelpText = errorsContract.HelpText
+            };
+        }
+
+        private static OctopusValidationException CreateOctopusValidationException(int statusCode, string body)
+        {
+            var errorsContract = OctopusErrorsContractFromBody(body);
+
+            return new OctopusValidationException(
+                statusCode,
+                errorsContract.ErrorMessage,
+                errorsContract.Errors,
+                errorsContract.Details)
+            {
+                HelpText = errorsContract.HelpText
+            };
+        }
+
+        private static OctopusServerException CreateOctopusServerException(int statusCode, string body)
+        {
+            var errorsContract = OctopusErrorsContractFromBody(body);
+            var fullDetails = errorsContract.ErrorMessage;
+            if (!string.IsNullOrWhiteSpace(errorsContract.FullException))
             {
+                fullDetails =
+                    $"Octopus Server returned an error: {errorsContract.ErrorMessage} {Environment.NewLine}Server exception: {Environment.NewLine}{errorsContract.FullException}{Environment.NewLine} ----------------------- {Environment.NewLine}";
             }
 
-            return new OctopusServerException(statusCode, fullDetails) {HelpText = helpText};
+            return new OctopusServerException(statusCode, fullDetails)
+            {
+                HelpText = errorsContract.HelpText
+            };
         }
 
-        static string GetErrorMessage(string body)
+        internal static OctopusErrorsContract OctopusErrorsContractFromBody(string body)
         {
+            OctopusErrorsContract result = null;
             string errorMessage;
+
             try
             {
-                var errors = JsonConvert.DeserializeObject<OctopusErrorsContract>(body);
-                errorMessage = errors.ErrorMessage;
+                result = JsonConvert.DeserializeObject<OctopusErrorsContract>(body);
+                errorMessage = result?.ErrorMessage ?? body;
+            }
+            catch (Exception jsonEx) when (jsonEx is ArgumentNullException or JsonSerializationException)
+            {
+                errorMessage = $"JSON deserialize error for {body} :: {jsonEx.Message}";
             }
-            catch
+            catch (Exception ex)
             {
-                errorMessage = body;
+                errorMessage = $"Unexpected json deserialization error for {body} :: {ex.Message}";
             }
-            return errorMessage;
+
+            return new OctopusErrorsContract
```

Comment:
The sensible defaults are in place to prevent null exception when constructing `octopus exception` (IE: OctopusValidationException)